### PR TITLE
fix(jsonrpc): update JSON-RPC Gateway to support headers and improve client registry

### PR DESCRIPTION
**Pull request by Ja7ad**
## Description

Update jrpc-gateway to v0.3.1 with new changes:

- update server and remove http server
- add graceful shutdown jrpc server
- remove register services pointer
- support metadata.MD

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -24,7 +24,7 @@ devtools:
 	go install github.com/bufbuild/buf/cmd/buf@v1.32
 	go install mvdan.cc/gofumpt@latest
 	go install github.com/rakyll/statik@v0.1.7
-	go install github.com/pacviewer/jrpc-gateway/protoc-gen-jrpc-gateway@v0.1.7
+	go install github.com/pacviewer/jrpc-gateway/protoc-gen-jrpc-gateway@latest
```
Comment (b00f) [inline Makefile:27]: Please specify version to avoid compiling unstable version

**Review by b00f — CHANGES_REQUESTED** [review]

File: www/jsonrpc/server.go [hunk 2]
```diff
@@ -50,32 +48,18 @@ func (s *Server) StartServer(grpcServer string) error {
...
 
-	blockchainService := pactus.NewBlockchainJsonRpcService(blockchain)
-	networkService := pactus.NewNetworkJsonRpcService(network)
-	transactionService := pactus.NewTransactionJsonRpcService(transaction)
-	walletService := pactus.NewWalletJsonRpcService(wallet)
-
-	jgw := jrpc.NewServer()
-	jgw.RegisterServices(&blockchainService, &networkService, &transactionService, &walletService)
+	server := jrpc.NewServer()
+	server.RegisterServices(blockchainService, networkService, transactionService, walletService)
```
Comment (b00f) [inline www/jsonrpc/server.go:57]: Can we merged this line into above register functions. We may pass  `server` to the `RegisterBlockchainJsonRpcService`?

**Review by b00f — CHANGES_REQUESTED** [review]

**Comment by b00f** [comment]
@Ja7ad `make check` helps to see the linting issues.

File: www/jsonrpc/server.go (hunk 2 above)
Comment (Ja7ad) [reply www/jsonrpc/server.go:57]: Registers create handers for server and server add this handlers to http.ServerMux

**Comment by Ja7ad** [comment]
> @Ja7ad `make check` helps to see the linting issues.

fixed.

**Review by b00f — APPROVED** [review]

**Pull request merged**